### PR TITLE
fix(recording): support long room tokens for uploads

### LINKED ISSUE #18458: Recording chunked upload would fail with token length > 13
But [tokens can be up to 30 characters long](https://github.com/nextcloud/spreed/blob/817b1280fd732cc1d9fe3d48f2420c63cdb85481/lib/Manager.php#L1483-L1492), no? Or am I misunderstanding the entropy parameter? :thinking:

_Originally posted by @danxuliu in https://github.com/nextcloud/spreed/pull/18449#discussion_r3481277576_

### PR DESCRIPTION
**Pull request by pollychen-lab**
## ☑️ Resolves

- Fix #18458

### AI (if applicable)

- [x] The content of this PR was partly or fully generated using AI

I used AI assistance to help draft and refine the code changes and regression tests in this pull request. I reviewed the final diff, verified the boundary behavior around the app-config key length, and rewrote this PR description myself.

### 🚧 Tasks

- [x] Keep recording upload app-config keys within Nextcloud's 64-character limit for long room tokens
- [x] Preserve the existing full SHA-1 hash for normal short-token cases
- [x] Add regression coverage for both boundary cases

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not possible
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

**Comment by nickvergessen** [comment]
- As per our AI policy, you need to expose the used AI model on commit: https://github.com/nextcloud/.github/blob/master/AI_POLICY.md#disclosure
- Also the communication (PR description) should be manual and you simply replaced the used template with your AI block: https://github.com/nextcloud/.github/blob/master/AI_POLICY.md#human-written-communication
- AI should not be used to work on "good first issues": https://github.com/nextcloud/.github/blob/master/AI_POLICY.md#beginner-issues

File: lib/Service/RecordingService.php [hunk 1]
```diff
@@ -55,6 +55,7 @@ class RecordingService {
 
 	public const APPCONFIG_PREFIX = 'recording/';
 	public const APPCONFIG_UPLOAD_PREFIX = 'recupload/';
+	private const APPCONFIG_KEY_MAX_LENGTH = 64;
```
Comment (nickvergessen) [inline lib/Service/RecordingService.php]:
```suggestion
	public const APPCONFIG_KEY_MAX_LENGTH = 64;
```
